etc/spec/ARCHITECTURE.md

```diff
@@ -119,12 +109,44 @@ The server can publish the following messages:
   4. `datetime_start`: A unix epoch in ms which indicates the datetime the
      message started being typed.
 
+The real-time server exposes an HTTP endpoint at `https://ting.gr/api/v1` with the following
+URLs:
+
+* A PATCH operation on `/message`: Sends an update on the message with an id equal to <messageid>.
+  The message can be sent from the user to a channel or to another user directly.
+  Takes five parameters:
+
+  1. `type`: A string which is either `channel` or `user`, indicating whether
+     the recipient of the message is a channel or a user.
+  2. `target`: The name of the channel or the user we wish to send the message
+     to.
+  3. `text`: The text of the message.
+  4. `typing`: A boolean that indicates if the message is currently
+     being typed or not.
+  5. `messageid`: An integer which is the unique id of the message being sent.
+
+     * There is no response for this operation.
```

Comment:
Also please clarify what happens when there is some validation problem with the data. What sort of response should be expected?